storybookjs/storybook
# Preact: Fix hooks when used in stories, preact-kitchen-sink

Issue: #13928.

Also resolves #11145 as it fixes Preact 10.x snapshots.

## What I did

- Upgraded dev version of preact to 10.x on the preact kitchen sink app to match the dev dependency installed.
- Changed from `preact-render-to-json` to `preact-render-to-string` for rendering Jest snapshots in tests.
- Changed `client/preview/render.tsx` to render stories using `h()` so that hooks work.
- Changed `storyshots-core/src/frameworks/preact/renderTree.ts` to use `preact-render-to-string` with `pretty: '  '` to output sensible HTML for snapshots.

## How to test

- Is this testable with Jest or Chromatic screenshots? ✔ The stories stopped rendering when the `devDependencies` on Storybook were upgraded to `10.x` but the kitchen sink app was kept as `8.x`. Not sure if the preact kitchen sink app is currently being checked, but it will render now.
- Does this need a new example in the kitchen sink apps? ✔ The `Addons > Actions > Action only` story has been updated to use a state hook to demonstrate that issue #13928 is fixed.
- Does this need an update to the documentation? ❌ It should still work for existing apps that pull `@storybook/preact` from npm since the runtime version of Storybook uses the consumer's version of preact, whichever is installed.
